R/config.R

```diff
@@ -0,0 +1,97 @@
+#  (c) Copyright 2023 Palantir Technologies Inc. All rights reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+#' @keywords internal
+DEFAULT_CONFIG_DIR <- file.path(path.expand("~"), ".foundry")
+
+#' @keywords internal
+ALIASES_FILE <- "aliases.yml"
+
+#' @keywords internal
+get_config_from_env <- function(name, default = NULL) {
+  value <- Sys.getenv(name, unset = NA)
+  if (is.na(value)) {
+    return(default)
+  }
+  return(value)
+}
+
+#' Loads a config from an environment variable with format `FOUNDRY_CONFIG_KEY`
+#' or from an option with format `foundry.config.key`.
+#'
+#' @keywords internal
+get_config <- function(name, default = NULL) {
+  # 1. Check for an environment variable
+  environment_variable <- sprintf("FOUNDRY_%s", toupper(gsub("\\.", "_", name)))
+  value <- get_config_from_env(environment_variable)
+  if (!is.null(value)) {
+    return(value)
+  }
+  # 2. Check for an option
+  option_variable <- sprintf("foundry.%s", name)
+  value <- getOption(option_variable)
```

Comment:
How are options set? If there are on individual sessions, think they should take higher priority compared to environment variables?

Reply:
Yes they're per R session, I'll revert the order

Reply:
(they're set by calling `options(foundry.hostname = "example.palantirfoundry.com"`)